pandas/core/arrays/base.py

```diff
@@ -655,15 +657,69 @@ def copy(self, deep=False):
         raise AbstractMethodError(self)
 
     # ------------------------------------------------------------------------
-    # Block-related methods
+    # Printing
     # ------------------------------------------------------------------------
+    def __repr__(self):
+        from pandas.io.formats.printing import format_object_summary
+
+        template = (
+            u'{class_name}'
+            u'{data}\n'
+            u'Length: {length}, dtype: {dtype}'
+        )
+        # the short repr has no trailing newline, while the truncated
+        # repr does. So we include a newline in our template, and strip
+        # any trailing newlines from format_object_summary
+        data = format_object_summary(self, self._formatter(), name=False,
+                                     trailing_comma=False).rstrip()
```

Comment:
I really find this ``trailing_comma`` argument bothersome. maybe it makes sense to name this like 'multi-line' or something, its seems that we built the ``format_object_summary`` to be for Index (which is true), then extended this (also true), so maybe it needs minor adjustment *inside* the method, rather than an option like this.

Reply:
I've removed `trailing_comma` in favor of a `.strip(', \n')` after formatting in a3b1c92347b469740a7d7ad20f9cb97b357086dd. Don't really have a preference for which we do.